effect-dynamodb@1.9.0 published with unresolved `workspace:` dep on @effect-dynamodb/schema (uninstallable by consumers)

## Summary

`effect-dynamodb@1.9.0` was published with an **unresolved `workspace:` protocol** in its dependencies:

```jsonc
// npm view effect-dynamodb@1.9.0 dependencies
{
  "@aws-sdk/client-dynamodb": "^3.700.0",
  "@aws-sdk/util-dynamodb": "^3.700.0",
  "@effect-dynamodb/schema": "workspace:^"   // <-- not rewritten at publish time
}
```

`workspace:` only resolves inside the originating monorepo. In any external project the dependency cannot be resolved, so the package is effectively **uninstallable for all consumers**.

## Reproduction

In a project that depends on `effect-dynamodb@^1.9.0` (pnpm):

```
ERR_PNPM_WORKSPACE_PKG_NOT_FOUND  In : "@effect-dynamodb/schema@workspace:^" is in the
dependencies but no package named "@effect-dynamodb/schema" is present in the workspace

This error happened while installing the dependencies of effect-dynamodb@1.9.0
```

npm/yarn consumers would hit the equivalent unresolved-specifier failure.

## Root cause

The `workspace:` protocol is meant to be rewritten to a concrete range at publish time (e.g. `workspace:^` → `^1.9.0`). `pnpm publish` / `pnpm pack` do this automatically; a plain `npm publish` (or publishing the raw `package.json`) does not. The `1.9.0` tarball shipped the literal `workspace:^`.

## Fix

Republish a patched version (e.g. `1.9.1`) using `pnpm publish` (or `pnpm pack` + publish the generated tarball) so the manifest resolves to a real range:

```jsonc
"@effect-dynamodb/schema": "^1.9.0"
```

Worth double-checking `@effect-dynamodb/schema@1.9.0` itself and any other workspace cross-deps in the same release, and consider a CI guard that greps the packed tarball for `workspace:` before publishing.

## Workaround for consumers (until 1.9.1)

pnpm override pinning the leaked spec to the real version:

```yaml
# pnpm-workspace.yaml
overrides:
  effect-dynamodb>@effect-dynamodb/schema: 1.9.0
```

## Affected

- effect-dynamodb 1.9.0 (only 1.9.x release currently published)

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -98,9 +98,26 @@ jobs:
           node <<'EOF'
           const { execSync } = require('child_process');
           const targets = require('/tmp/publish.json');
+          execSync('mkdir -p /tmp/pkgs');
           for (const { name, version, dir } of targets) {
-            console.log(`Publishing ${name}@${version} from ${dir}`);
-            execSync('npm publish --provenance --access public', { cwd: dir, stdio: 'inherit' });
+            console.log(`Packing ${name}@${version} from ${dir}`);
+            // `pnpm pack` rewrites the `workspace:` protocol in dependencies AND
+            // peerDependencies to concrete ranges; a plain `npm publish` does NOT,
+            // which shipped an uninstallable `workspace:^` in 1.9.0 (issue #64).
+            // We publish the packed tarball so the resolved manifest is what ships.
+            const out = execSync('pnpm pack --pack-destination /tmp/pkgs', { cwd: dir, encoding: 'utf8' });
+            const tgz = out.trim().split('\n').pop().trim();
+            // Guard: refuse to publish if any unresolved `workspace:` spec remains.
+            const pkg = JSON.parse(execSync(`tar -xzOf ${tgz} package/package.json`, { encoding: 'utf8' }));
+            for (const field of ['dependencies', 'peerDependencies', 'optionalDependencies']) {
+              for (const [dep, spec] of Object.entries(pkg[field] || {})) {
+                if (String(spec).startsWith('workspace:')) {
+                  throw new Error(`Refusing to publish ${name}: ${field}.${dep} = "${spec}" — unresolved workspace: protocol (issue #64).`);
+                }
+              }
+            }
+            console.log(`Publishing ${tgz}`);
+            execSync(`npm publish ${tgz} --provenance --access public`, { stdio: 'inherit' });
           }
           EOF
 

--- a/packages/effect-dynamodb-geo/CHANGELOG.md
+++ b/packages/effect-dynamodb-geo/CHANGELOG.md
@@ -1,5 +1,24 @@
 # @effect-dynamodb/geo
 
+## 1.9.1
+
+### Patch Changes
+
+- fix(release): resolve `workspace:` protocol at publish time (closes #64).
+
+  `1.9.0` shipped with an unresolved `workspace:` spec (`effect-dynamodb`'s
+  `dependencies."@effect-dynamodb/schema": "workspace:^"`, and `@effect-dynamodb/geo`'s
+  `peerDependencies.effect-dynamodb`), making `effect-dynamodb@1.9.0` uninstallable for
+  consumers. Root cause: `release.yml` published via `npm publish`, which does not
+  rewrite the `workspace:` protocol.
+
+  The publish step now packs each package with `pnpm pack` (which rewrites `workspace:`
+  in `dependencies` and `peerDependencies` to concrete ranges) and publishes the
+  resulting tarball via `npm publish` (preserving OIDC Trusted Publishing + provenance),
+  with a guard that refuses to publish if any `workspace:` spec remains in the packed
+  manifest. No runtime/API changes — 1.9.1 republishes 1.9.0 with correctly resolved
+  dependency ranges.
+
 ## 1.9.0
 
 ### Minor Changes

--- a/packages/effect-dynamodb-geo/package.json
+++ b/packages/effect-dynamodb-geo/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@effect-dynamodb/geo",
-  "version": "1.9.0",
+  "version": "1.9.1",
   "description": "Geospatial index and search for effect-dynamodb using H3 hexagonal grid",
   "type": "module",
   "license": "MIT",

--- a/packages/effect-dynamodb/CHANGELOG.md
+++ b/packages/effect-dynamodb/CHANGELOG.md
@@ -1,5 +1,27 @@
 # effect-dynamodb
 
+## 1.9.1
+
+### Patch Changes
+
+- fix(release): resolve `workspace:` protocol at publish time (closes #64).
+
+  `1.9.0` shipped with an unresolved `workspace:` spec (`effect-dynamodb`'s
+  `dependencies."@effect-dynamodb/schema": "workspace:^"`, and `@effect-dynamodb/geo`'s
+  `peerDependencies.effect-dynamodb`), making `effect-dynamodb@1.9.0` uninstallable for
+  consumers. Root cause: `release.yml` published via `npm publish`, which does not
+  rewrite the `workspace:` protocol.
+
+  The publish step now packs each package with `pnpm pack` (which rewrites `workspace:`
+  in `dependencies` and `peerDependencies` to concrete ranges) and publishes the
+  resulting tarball via `npm publish` (preserving OIDC Trusted Publishing + provenance),
+  with a guard that refuses to publish if any `workspace:` spec remains in the packed
+  manifest. No runtime/API changes — 1.9.1 republishes 1.9.0 with correctly resolved
+  dependency ranges.
+
+- Updated dependencies
+  - @effect-dynamodb/schema@1.9.1
+
 ## 1.9.0
 
 ### Minor Changes

--- a/packages/effect-dynamodb/package.json
+++ b/packages/effect-dynamodb/package.json
@@ -1,6 +1,6 @@
 {
   "name": "effect-dynamodb",
-  "version": "1.9.0",
+  "version": "1.9.1",
   "description": "Effect TS ORM for DynamoDB — schema-driven entity modeling, single-table design, composite keys, transactions, and Stream-based pagination",
   "type": "module",
   "license": "MIT",

--- a/packages/language-service/CHANGELOG.md
+++ b/packages/language-service/CHANGELOG.md
@@ -1,5 +1,24 @@
 # @effect-dynamodb/language-service
 
+## 1.9.1
+
+### Patch Changes
+
+- fix(release): resolve `workspace:` protocol at publish time (closes #64).
+
+  `1.9.0` shipped with an unresolved `workspace:` spec (`effect-dynamodb`'s
+  `dependencies."@effect-dynamodb/schema": "workspace:^"`, and `@effect-dynamodb/geo`'s
+  `peerDependencies.effect-dynamodb`), making `effect-dynamodb@1.9.0` uninstallable for
+  consumers. Root cause: `release.yml` published via `npm publish`, which does not
+  rewrite the `workspace:` protocol.
+
+  The publish step now packs each package with `pnpm pack` (which rewrites `workspace:`
+  in `dependencies` and `peerDependencies` to concrete ranges) and publishes the
+  resulting tarball via `npm publish` (preserving OIDC Trusted Publishing + provenance),
+  with a guard that refuses to publish if any `workspace:` spec remains in the packed
+  manifest. No runtime/API changes — 1.9.1 republishes 1.9.0 with correctly resolved
+  dependency ranges.
+
 ## 1.9.0
 
 ### Minor Changes

--- a/packages/language-service/package.json
+++ b/packages/language-service/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@effect-dynamodb/language-service",
-  "version": "1.9.0",
+  "version": "1.9.1",
   "description": "TypeScript Language Service Plugin for effect-dynamodb — shows DynamoDB operations on hover",
   "license": "MIT",
   "author": "Justin Menga",

--- a/packages/schema/CHANGELOG.md
+++ b/packages/schema/CHANGELOG.md
@@ -1,5 +1,24 @@
 # @effect-dynamodb/schema
 
+## 1.9.1
+
+### Patch Changes
+
+- fix(release): resolve `workspace:` protocol at publish time (closes #64).
+
+  `1.9.0` shipped with an unresolved `workspace:` spec (`effect-dynamodb`'s
+  `dependencies."@effect-dynamodb/schema": "workspace:^"`, and `@effect-dynamodb/geo`'s
+  `peerDependencies.effect-dynamodb`), making `effect-dynamodb@1.9.0` uninstallable for
+  consumers. Root cause: `release.yml` published via `npm publish`, which does not
+  rewrite the `workspace:` protocol.
+
+  The publish step now packs each package with `pnpm pack` (which rewrites `workspace:`
+  in `dependencies` and `peerDependencies` to concrete ranges) and publishes the
+  resulting tarball via `npm publish` (preserving OIDC Trusted Publishing + provenance),
+  with a guard that refuses to publish if any `workspace:` spec remains in the packed
+  manifest. No runtime/API changes — 1.9.1 republishes 1.9.0 with correctly resolved
+  dependency ranges.
+
 ## 1.9.0
 
 ### Minor Changes

--- a/packages/schema/package.json
+++ b/packages/schema/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@effect-dynamodb/schema",
-  "version": "1.9.0",
+  "version": "1.9.1",
   "description": "Pure Effect Schema-driven entity/aggregate modeling and relationship derivation for effect-dynamodb — importable without @aws-sdk",
   "type": "module",
   "license": "MIT",

